PowerManager/PowerUtils.cpp

```diff
@@ -151,3 +151,65 @@ WakeupSrcType PowerUtils::conv(const std::string& wakeupSrc)
         return WakeupSrcType::WAKEUP_SRC_UNKNOWN;
     }
 }
+
+// IMPORTANT: Keep this conversion in sync with the text alias provided for
+//            PowerState in IPowerManager.h interface file
+PowerState PowerUtils::convPowerState(const std::string& powerState)
+{
+    PowerState state = PowerState::POWER_STATE_UNKNOWN;
+    std::string stateStr = powerState;
+    std::transform(stateStr.begin(), stateStr.end(), stateStr.begin(), ::toupper);
+
+    if (stateStr == "OFF") {
+        state = PowerState::POWER_STATE_OFF;
+    } else if (stateStr == "STANDBY") {
+        state = PowerState::POWER_STATE_STANDBY;
+    } else if (stateStr == "ON") {
+        state = PowerState::POWER_STATE_ON;
+    } else if (stateStr == "LIGHT_SLEEP") {
+        state = PowerState::POWER_STATE_STANDBY_LIGHT_SLEEP;
+    } else if (stateStr == "DEEP_SLEEP") {
+        state = PowerState::POWER_STATE_STANDBY_DEEP_SLEEP;
+    } else {
+        LOGERR("Unknown PowerState string:[%s]", powerState.c_str());
+    }
+    return state;
+}
+
+std::string PowerUtils::readFromFile(const std::string& path)
```

Comment:
why you need this extra function?


you can build some simple logic:

during bootup do following create a variable for dbfile path:
1) check if /tmp/uimgr_settings.bin present dbfile path = /tmp/uimgr_settings.bin
2) if /tmp/uimgr_settings.bin not present dbfile path = /opt/uimgr_settings.bin and cp /opt/uimgr_settings.bin to /tmp/uimgr_settings.bin